src/util/span.h

```diff
@@ -1,6 +1,14 @@
 #pragma once
 
+#if __has_include(<span>)
```

Comment:
Unfortunately this does not help, because we need the c++20 include <version> which is not available in Ubuntu Focal

Reply:
Right, but why would `__cpp_lib_span` be undefined if the include is not available?

Reply:
Oh sorry, GitHub eat the  angle brackets of my comments above. Fixed.

`__cpp_lib_span` is defined in `<version>` or `<span>`, but `#include <version>` will also not compile on Ubuntu Focal. 
We may consider an additional check for `__cpp_lib_span`, but that is IMHO only clutter.

Reply:
> Oh sorry, GitHub eat the angle brackets of my comments above. Fixed.

Ah, yes now your comment makes sense. 

> `__cpp_lib_span` is defined in `<version>` or `<span>`, but `#include <version>` will also not compile on Ubuntu Focal.

Are you sure? IIUC `<span>` is only included in gcc10 but `<version>` is included in gcc9 and thus should be available on Ubuntu Focal (assuming gcc is properly configured).

Reply:
Digging into the gcc9 source tree, the `<version>` header seems to be available indeed. https://gcc.gnu.org/git/?p=gcc.git;a=blob_plain;f=libstdc%2B%2B-v3/include/std/version;hb=refs/heads/releases/gcc-9

Reply:
Oh yes, it is available with libstdc++9.

But it should more robust to not require a new c++20 feature to check for another feature from a pre c++20 compiler. 

But since it will work in our case, I don't mind. 
What do you think, should I add your change or not?

Reply:
> But it should more robust to not require a new c++20 feature to check for another feature from a pre c++20 compiler.

I agree, but we agreed that using the 50% of C++20 that are available in gcc9 are good to use. 

> What do you think, should I add your change or not?

I feel like `__has_include` is more hacky than checking for a specific `#define`, which is why I personally prefer that, but your concern is also valid. So I don't really care if you accept my change or not.

Reply:
OK than please merge it as it is and safe us form creating another commit.